docs/en/advanced_guides/add_modules.md

```diff
@@ -232,3 +166,60 @@ Then you need to modify the `loss_decode` field in the head.
 ```python
 loss_decode=dict(type='MyLoss', loss_weight=1.0))
 ```
+
+## Develop new segmentors
+
+The segmentor is an algorithmic architecture in which users can customize their own algorithms by adding their own components and defining the logic of algorithm execution. Please refer to [model document](https://github.com/open-mmlab/mmsegmentation/blob/1.x/docs/en/advanced_guides/models.md) for more details.
+
+Here we show how to develop a new segmentor.
+
+1. Create a new file `mmseg/models/segmentors/my_segmentor.py`.
+
+   ```python
+   from typing import Dict, Optional, Union
+
+   import torch
+
+   from mmseg.registry import MODELS
+   from mmseg.models import BaseSegmentor
+
+   @MODELS.register_module()
+   class MySegmentor(BaseSegmentor):
+
+       def __init__(self, **kwargs):
+           super().__init__(**kwargs)
+
+       def train_step(self,
+                   data: Union[dict, tuple, list],
+                   optim_wrapper: OptimWrapper) -> Dict[str, torch.Tensor]:
+           pass
+
+       def val_step(self,
+                   data: Union[tuple, dict, list]) -> list:
+           pass
+
+       def test_step(self,
+                   data: Union[tuple, dict, list]) -> list:
+           pass
```

Comment:
`BaseModel` in MMEngine implements xx_step, and mmseg didn't rewrite them.

I suggest that we should tell users what  they should inherit from or rewrite, if they would like to add a segmentor

https://github.com/open-mmlab/mmengine/blob/ca0364bbc3a6ac9132f5d499de190f7e733383ca/mmengine/model/base_model/base_model.py#L84